src/gt4py/next/program_processors/runners/dace_iterator/workflow.py

```diff
@@ -204,8 +206,12 @@ def convert_args(
     on_gpu = True if device == core_defs.DeviceType.CUDA else False
 
     def decorated_program(
-        *args, offset_provider: dict[str, common.Connectivity | common.Dimension]
+        *args, offset_provider: dict[str, common.Connectivity | common.Dimension], out: Any = None
     ):
+        if out is not None:
+            args = (*args, out)
+        if len(sdfg.arg_names) > len(args):
+            args = (*args, *gtfn.iter_size_args(args))
         if sdfg_program._lastargs:
```

Comment:
For everybody's knowledge, `sdfg_program._lastargs` is the list of program arguments in the order expected by dace. I use the array `inp.sdfg_arg_position` to map the GT4Py position to the position in the DaCe expected arguments list. Therefore, as I verbally mentioned once or twice, the dace backend will reorder the program arguments a second time. Is it possible or desirable to let the backend setup the expected order?
Besides, as you can see in the lines below, we distinguish scalar arguments from field arguments: scalar arguments are written always, while field arguments are written only when the field is reallocated. Is there a way (an execution path) to easily detect when the fields are re-allocated?

Reply:
If we can hand over recognizing the generated "out" keyword argument as well as the implicit domain size arguments to DaCe in some way, it would probably be beneficial. I don't know whether that's possible though.